spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -4,19 +4,35 @@ import com.datastax.driver.core._
 import com.datastax.spark.connector.types.ColumnType
 import org.apache.spark.Logging
 
+/**
+ * Class for binding row-like objects into prepared statements. prefixVals
+ * is used for binding constant values into each bound statement. This supports parametrized
+ * .where clauses in [[com.datastax.spark.connector.rdd.CassandraJoinRDD]]
+ */
 private[connector] class BoundStatementBuilder[T](
     val rowWriter: RowWriter[T],
     val preparedStmt: PreparedStatement,
-    val protocolVersion: ProtocolVersion) extends Logging {
+    val protocolVersion: ProtocolVersion,
+    val prefixVals: Seq[Any] = Seq.empty) extends Logging {
 
   private val columnNames = rowWriter.columnNames.toIndexedSeq
   private val columnTypes = columnNames.map(preparedStmt.getVariables.getType)
   private val converters = columnTypes.map(ColumnType.converterToCassandra(_)(protocolVersion))
   private val buffer = Array.ofDim[Any](columnNames.size)
 
+  private val prefixTypes = (0 until prefixVals.length).map(preparedStmt.getVariables.getType)
+  private val prefixConverted = for {
```

Comment:
@cin I don't think using view in this case is a good idea. This is going to be used for every processed row so we need to tune the performance as much as possible. If if used view, it would look for the converter and then convert each prefix val for each row.